### PR TITLE
Feature/refactoring c make lists for unit tests

### PR DESCRIPTION
**Pull request by BSolonenko**
This PR is **[ready]** for review.

### Risk
This PR makes **[no]** API changes.

### Summary
These changes include low_voltage_signals_handler_test into a build. CMakeLists.txt for creating this test was refactored according to the changes of BSolonenko.

##### Breaking Changes
- included the folder "appMain/test" into the build
- changed appropriate CMakeLists.txt 
- added some changes into transport_manager/CMakeLists.txt to fix build errors
- changed function in sources.cmake "create test"

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/3rd_party/expat-2.1.0/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,113 @@
+
```
Comment (AKalinich-Luxoft) [inline src/3rd_party/expat-2.1.0/CMakeLists.txt]: @BSolonenko I think our copyright should be added instead

File: src/3rd_party/log4cxx/Linux.cmake [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# Copyright (c) 2014, Ford Motor Company All rights reserved.
```
Comment (AKalinich-Luxoft) [inline src/3rd_party/log4cxx/Linux.cmake]: @BSolonenko 2018

File: src/3rd_party/log4cxx/QNX7.cmake [hunk 3]
```diff
@@ -0,0 +1,59 @@
+# Copyright (c) 2014, Ford Motor Company All rights reserved.
```
Comment (AKalinich-Luxoft) [inline src/3rd_party/log4cxx/QNX7.cmake]: @BSolonenko 2018

File: src/3rd_party/set_3rd_party_paths.cmake [hunk 4]
```diff
@@ -1,48 +1,46 @@
-# Copyright (c) 2014, Ford Motor Company
-# All rights reserved.
+# Copyright (c) 2014, Ford Motor Company All rights reserved.
```
Comment (AKalinich-Luxoft) [inline src/3rd_party/set_3rd_party_paths.cmake]: @BSolonenko 2018

File: tools/cmake/helpers/sources.cmake [hunk 5]
```diff
@@ -134,12 +134,22 @@ function(collect_sources SOURCES PATHS)
...
+                                    "${COMPONENTS_DIR}/include")
+
+  target_sources(${NAME} PRIVATE
+                 ${CMAKE_SOURCE_DIR}/src/components/test_main.cc)
+
+  target_link_libraries(${NAME} PUBLIC utils::utils gmock)
+
   add_test(NAME ${NAME}
-  COMMAND ${NAME} --gtest_output=xml:${CMAKE_BINARY_DIR}/test_results/)
+           COMMAND ${NAME} --gtest_output=xml:${CMAKE_BINARY_DIR}/test_results/)
```
Comment (SNiukalov) [inline tools/cmake/helpers/sources.cmake]: Please remove the space.

Comment (SNiukalov) [inline tools/cmake/helpers/sources.cmake]: Please remove the space

**Pull request merged**